pkg/api/validate/api.go

```diff
@@ -63,16 +69,12 @@ func (v *APIValidator) validateUpdateContainerService(cs, oldCs *api.OpenShiftMa
 			}
 		}
 	}
+	old.Properties.AuthProfile.IdentityProviders = cs.Properties.AuthProfile.IdentityProviders
 
 	if !reflect.DeepEqual(cs, old) {
 		// TODO: this is a hack because we're using deep.Equal.  To fix properly
 		// we'd probably need to implement our own deep.Equal.
 		csCopy := cs.DeepCopy()
-		if csCopy.Properties.AuthProfile.IdentityProviders[0].Provider.(*api.AADIdentityProvider).Secret !=
-			old.Properties.AuthProfile.IdentityProviders[0].Provider.(*api.AADIdentityProvider).Secret {
-			csCopy.Properties.AuthProfile.IdentityProviders[0].Provider.(*api.AADIdentityProvider).Secret = "<hidden 1>"
-			old.Properties.AuthProfile.IdentityProviders[0].Provider.(*api.AADIdentityProvider).Secret = "<hidden 2>"
-		}
```

Comment:
I think removing these lines just removes the secret redaction in logging, are we sure we want to do this?

Reply:
Same question. What is the reasoning to remove it?